### PR TITLE
Add support for chatting with models hosted on Databricks

### LINKED ISSUE #55: Explore databricks backend

### PR DESCRIPTION
**Pull request by atheriel**
This commit adds support for chatbots that work with [models hosted on Databricks][0].

According to their documentation, Databricks models can be queried by the OpenAI Python SDK, so in theory they would be compatible with our OpenAI provider. But in practice there are enough differences that a new provider is necessary.

There are other limitations, too: structured outputs are in public preview, and there is no image support.

Finally, I've found that tool calling support (also in public preview) is very limited, too limited for `elmer` to work properly at all:

```r
> chat <- chat_databricks()
> chat$register_tool(tool(function() "2024-01-01", "Gets the current date"))
Using model = "databricks-dbrx-instruct".
> chat$chat("What's the current date in YMD format?")
Error in `req_perform_connection()` at elmer/R/httr2.R:36:3:
! HTTP 400 Bad Request.
• Bad request: Model does not support continuing the chat past the first tool call
Show Traceback
> chat
<Chat turns=2 tokens=726/11>
── user ────────────────────────────────────────────────────────────────────────────────────────
What's the current date in YMD format?
── assistant ───────────────────────────────────────────────────────────────────────────────────
[tool request (call_93d2e2fe-1a2e-4ef4-80df-4feaed245569)]: tool_001()
```

Databricks has announced that this limition will be lifted [once tool calling is out of public preview][1], so I suppose we'll have to cross our fingers that this will start working at that point.

Unit tests and docs are included.

Closes #55.

[0]: https://docs.databricks.com/en/machine-learning/model-serving/score-foundation-models.html
[1]: https://docs.databricks.com/en/machine-learning/model-serving/function-calling.html#limitations

### CODE REVIEW

File: tests/testthat/test-provider-databricks.R [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  resp <- coro::collect(chat$stream("What is 1 + 1?"))
+  expect_match(paste0(unlist(resp), collapse = ""), "2")
+})
+
+# Common provider interface -----------------------------------------------
+
+test_that("defaults are reported", {
+  expect_snapshot(. <- chat_databricks())
+})
+
```
[the remark this answers is not in the document]
Comment (atheriel) [reply tests/testthat/test-provider-databricks.R:25]: Do you want me to add them commented-out with an explanation as to what is broken? Or have tests with expected errors?

[the remark this answers is not in the document]
Comment (hadley) [reply tests/testthat/test-provider-databricks.R:25]: Yeah, just commented out with an explanation.

**Comment by hadley** [comment]
Shall we add this to the automated tests too?

**Comment by atheriel** [comment]
This PR only supports PAT authentication, which we could use for CI, but I'd rather land support for Databricks service principals first. I have a SP set up for CI use already. Can we hold off until then?

**Comment by hadley** [comment]
Yeah, sure thing.

**Pull request merged**